### DISCUSSION TITLE
API Route stack trace line numbers don't match /pages/api line numbers

### DISCUSSION THREAD

**Discussion by jcald1** [vercel/next.js#15920] [Help]
I'm using v9.3.5. The API route stack trace doesn't match the sources line number in the /pages/api directory

**Comment by AubreyHewes** [comment]
@jcald1 do you have more information?

There are so many things that can break your line numbers (for example pre-compiling your source code; babel/ts).. have you tried debugging your api route?

**Comment by jcald1** [comment]
@AubreyHewes , I haven't done anything out of the ordinary. It's just a basic API route in a simple app.  I was able to figure out the underlying issue, but I can't tell what is causing the line numbers not to match.

If you clone https://github.com/datadeckllc/datadeck and run it without setting up the Go component, then upload a file on the main page, it will trigger an error.

On the server log, you can see: 

```
Finished Calling SOS CA. {
  err: {
    name: 'Error',
    message: 'Error calling SOS CA',
    stack: 'Error: Error calling SOS CA\n' +
      '    at /home/user/projects/datadeckllc/datadeck/.next/server/static/development/pages/api/seed-data.js:331:87\n'
...
```

It's showing the line numbers from the `.next` directory, not the source directory.

**Comment by tettoffensive** [comment]
@jcald1 I'm constantly having this issue with debugging API routes. Often I can't place breakpoints in vscode, or if I do it breaks at some totally random spot (sometimes it will open the correct transpiled file and then I can just move the breakpoint to where it should be). 

The only chance I have is to stop my server, restart vscode, blow away the the .next directory, and restart. But as soon as I change some code, I have to do it all over again.

I feel like there must be some trick to it, otherwise, I feel like i'd see more complaints about it.

```json
{
      "type": "node",
      "request": "attach",
      "name": "Attach to Program",
      "skipFiles": [
        "<node_internals>/**"
      ],      
      "port": 9229
    },
```

```json
"dev": "NODE_OPTIONS='--inspect' next dev",
```
